frontend/src/store/modules/models.js

```diff
@@ -27,14 +27,16 @@ const getters = {
 };
 
 const actions = {
-  async getModels({ commit }) {
-    const models = await modelsApi.fetchModels();
-    commit('setModelList', models.sort((a, b) => (a.short_name.toLowerCase() < b.short_name.toLowerCase() ? -1 : 1)));
+  async getModels({ commit, state }) {
+    if (state.modelList.length === 0) {
+      const models = await modelsApi.fetchModels();
+      commit('setModelList', models.sort((a, b) => (a.short_name.toLowerCase() < b.short_name.toLowerCase() ? -1 : 1)));
+    }
   },
   /* eslint-disable no-shadow */
-  async selectModel({ dispatch, commit, getters, state }, modelShortName) {
+  async selectModel({ dispatch, commit, getters }, modelShortName) {
     if (state.modelList.length === 0) {
-      await dispatch('getModels');
+    await dispatch('getModels');
```

Comment:
Not on the right computer right now so I haven't tested this. But how come the state can be omitted on line 37? Also the indentation seems off now on line 39.

Reply:
Good catch! It's a mistake - I had moved the condition check inside `getModels` above. I'll fix this on another branch.